### ISSUE TITLE
Change the way we deal with deleted parameters in config files

### ISSUE THREAD

**Issue by ShaiberAlon** [merenlab/anvio#1313]
Discussion here: https://github.com/merenlab/anvio/commit/b6dff45e66f3d8e3da3e4a72344a1332f87cef26#r36167134

## Details of the change:
1. We will make it so that deleted parameters are set to `None`.
2. To track the version of config files we will add a version parameter.
3. From here on, a config file with no `version` parameter will not be accepted (an error will be issued, saying either to manually re-enter the version parameter to the config (in case it was deleted by the user), or to use the migration script to migrate the config file.
3. Add a migration script for config file.

## Migration of config file
If no `version` parameter is found in the config file then it will be considered version zero.

The migration from `v0` to `v1` is easy, we will simply take the config file from the user, and will use the `dict` built-in function `update` to "update" the default config file, and then return that result as the new config file.

Here is a sketch of it in code:

```python
import argparse
import anvio.workflows as w
import anvio.filesnpaths as filesnpaths

args = argparse.Namespace(workflow=workflow_name)

workflow_module_dict = w.get_workflow_module_dict()
workflow_object = workflow_module_dict[workflow_name](args)

user_config_v1 = workflow_object.default_config
user_config_v1.update(user_config_v0)

filesnpaths.is_output_file_writable(file_path)
open(file_path, 'w').write(json.dumps(user_config_v1, indent=4))
```

### How to migrate?
I think that the migration should be done using `anvi-run-workflow` with a new `--migrate-config`, but we could also do it through `anvi-migrate-db` (and then we should maybe rename it to `anvi-migrate`).

### How would we know which workflow?
Notice that migration requires us to know the name of the workflow. There are two solutions for this:
1. Add another new parameter to the config file to specify the name of the workflow.
2. Have the user specify the name of the workflow when they provide a config file.

Adding a workflow_name parameter to config would allow any config with version 1 and above to be upgraded in mass (think `anvi-migrate *json`). But we wouldn't be able to do this for configs of `v0`, which means that we can't avoid having an option for the user to specify workflow name.

If we have the user specify the workflow name then they can still do `anvi-run-workflow -w contigs --migrate-config *json`, but they would need to make sure all those config files are truly for contigs workflows.

The need to specify the workflow name, makes me think the migration should be done using `anvi-run-workflow` and not using `anvi-migrate-db`. I think it would be more natural.

### Bottom line - what I plan for now

I will implement a script for migration in `sandbox` and later we can decide what we do:
`anvi-script-migrate-config-v0-v1`
